### PR TITLE
Updates blockType and viewType icons for the 'Web Stories List' block.

### PR DESCRIPTION
**Pull request by divyarajmasani**
## Summary

Updates icons for the blockType and layout selection buttons.

## User-facing changes

- Updates user facing panel and layout order.

- ### BlockType selection panel:
  - Before
Screenshot 2020-12-28 at 5 21 11 PM

  - Now
Screenshot 2020-12-25 at 12 47 51 PM

- ### Layout selection panel:
  - Before
Screenshot 2020-12-28 at 5 21 18 PM

  - Now
Screenshot 2020-12-28 at 5 29 37 PM

### CODE REVIEW

**Review by ankitrox — APPROVED** [review]
LGTM 👍

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (spacedmonkey) [inline assets/src/web-stories-block/block/components/storiesInspectorControls.js:52]: @swissspidy Could we be used styled-components in gutenberg. Does that add a lot of overhead to block editor?

Comment (spacedmonkey) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:52]: Doesnt it make more sense to use [edit.css](https://github.com/rtCamp/web-stories-wp/blob/1795867f6456d3db7dd1cb6c127fe22554f7f512/assets/src/web-stories-block/block/edit.css) -

File: assets/src/web-stories-block/block/icons.js
Comment (spacedmonkey) [inline assets/src/web-stories-block/block/icons.js:125]: Not sure I understand hardcoding SVG files like this. Elsewhere just have the svg as a [file](https://github.com/rtCamp/web-stories-wp/tree/1795867f6456d3db7dd1cb6c127fe22554f7f512/assets/src/edit-story/icons) and [import](https://github.com/rtCamp/web-stories-wp/blob/1795867f6456d3db7dd1cb6c127fe22554f7f512/assets/src/edit-story/elements/video/icon.js#L25) it.

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (spacedmonkey) [inline assets/src/web-stories-block/block/components/storiesInspectorControls.js:131]: Why does this open in a window / tab?

File: includes/Block/Web_Stories_Block.php
Comment (spacedmonkey) [inline includes/Block/Web_Stories_Block.php:185]: $rest_base != $post_type. 

This variable is confusing. 

You should have the rest base to look more like [this](https://github.com/google/web-stories-wp/blob/31dffad61c908082a721218f428b7a88ea0ea285/includes/Story_Post_Type.php#L505).

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:52]: I understand, it definitely not as light as using the CSS. On quick search I found some 30 styled components in our block source. Some these are components we used from the `dashboard` and have overridden the styles in our block as needed.

You may also want to take a look at the blocks PRs #188 and #190.

@swissspidy

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:131]: Should it open in the same tab as the editor? 
I though it might not be a good thing for the user to open the link in the same tab while they are editing it.

File: assets/src/web-stories-block/block/icons.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/icons.js:125]: our webpack build use `svgr/webpack`, which strips of the dimensions from the svg which are needed for these icons to be displayed on the block toolbar, without `width/height` the SVG spans full width and we don't have direct control via CSS either to limit the dimension without affecting other icons.

`svg/webpack` uses [config file](https://github.com/rtCamp/web-stories-wp/blob/feature/block-icons-update/.svgo.yml#L3), we wouldn't want to modify this here as it will affect the dashboard as well, so best option was to use them in JS as constants.

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (swissspidy) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:131]: New tab should be ok for now

File: assets/src/web-stories-block/block/edit.js
Comment (swissspidy) [inline assets/src/web-stories-block/block/edit.js:71]:
```suggestion
        instruction={__('Select a layout style', 'web-stories')}
```

File: assets/src/web-stories-block/block/icons.js
Comment (swissspidy) [reply assets/src/web-stories-block/block/icons.js:125]: I see what you mean. Could you please add a code comment in this file with this explanation? Perhaps with a TODO mark? It will make our lives as maintainers much easier.This way we can come back to it later and fix it.

----

Out of curiosity, have you tried something like this:

```js
import { ReactComponent as GridViewTypeIcon } from '../../icon.svg';
export const GRID_VIEW_TYPE_ICON = <GridViewTypeIcon width={12} height={14} />
```

(probably doesn't work, but worth giving it a shot)

File: includes/Block/Web_Stories_Block.php
Comment (swissspidy) [inline includes/Block/Web_Stories_Block.php:174]: 👍

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (swissspidy) [inline assets/src/web-stories-block/block/components/storiesInspectorControls.js:131]: Why is this a string and not a React component?

Should not be using `RawHTML` for this

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (swissspidy) [inline assets/src/web-stories-block/block/components/storiesInspectorControls.js:189]: Should be using a React component, not `RawHTML`

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (swissspidy) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:52]: Good points here. I see where both of you are coming from.

All the dashboard components rely on styled-components here, which is fine/expected at the moment.

That means styled-components will be already loaded and available, no matter what.

Hence it is fine to also use it here.

Should we ever use other components here (e.g. not relying on dashboard components), we might want to reconsider this.

For now I'd say it's OK though 👍

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (spacedmonkey) [inline assets/src/web-stories-block/block/components/storiesInspectorControls.js:134]:
```suggestion
  const archiveLink = (<a target="__blank" href={archiveURL}>{__(
    'View archive page',
    'web-stories'
  )}</a>);
```

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (spacedmonkey) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:189]:
```suggestion
                    show && < ArchiveLink  />
```

File: assets/src/web-stories-block/block/icons.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/icons.js:125]: I haven't tried with the dimension attrs, though did try with the `size` attribute. I'll update the PR if it works.

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js:134]: This results in error, instead need to use following
```suggestion
    const ArchiveLink = () => (
    <a target="__blank" href={archiveURL}>
      {__('View archive page', 'web-stories')}
    </a>
  );
```

This returns a component and gutenberg/react doesn't complain about it.

Found this: https://github.com/facebook/react/issues/13445

File: assets/src/web-stories-block/block/components/storiesInspectorControls.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/components/storiesInspectorControls.js]: updated to use component instead as suggested.

File: assets/src/web-stories-block/block/icons.js
Comment (divyarajmasani) [reply assets/src/web-stories-block/block/icons.js]: This worked, and I've updated the PR with the required changes.

**Review by swissspidy — APPROVED** [review]

**Comment by divyarajmasani** [comment]
As this branch / changes were stable and we are not expecting anymore changes for the icons. 
I'm keeping this aside as there are conflicts in rebase, will update this once we have #190 merged.

**Review by spacedmonkey — APPROVED** [review]
SGTM

**Review by swissspidy — APPROVED** [review]

**Review by swissspidy — APPROVED** [review]

**Pull request merged**
